django_tasks/backends/immediate.py

```diff
@@ -37,14 +37,21 @@ def enqueue(
         try:
             result = json_normalize(calling_task_func(*args, **kwargs))
             status = ResultStatus.COMPLETE
-        except Exception as e:
+        except BaseException as e:
             try:
                 result = exception_to_dict(e)
             except Exception:
                 logger.exception("Task id=%s unable to save exception", result_id)
                 result = None
+
+            # Use `.exception` to integrate with error monitoring tools (eg Sentry)
+            logger.exception("Task execution failed: %s", e)
             status = ResultStatus.FAILED
 
+            # If the user tried to terminate, let them
+            if isinstance(e, KeyboardInterrupt):
+                raise
```

Comment:
Similarly [inspired from db_worker](https://github.com/RealOrangeOne/django-tasks/blob/master/django_tasks/backends/database/management/commands/db_worker.py#L135-L137)

Reply:
I think this is fine. As part of #59 I'd like to add tests to make sure this all works correctly as we'd expect, but that can come later.